chain/client/src/client_actor.rs

```diff
@@ -1196,9 +1206,14 @@ impl ClientActorInner {
                 },
                 &|validator_signer| self.client.update_validator_signer(validator_signer),
             );
+
             if update_result.validator_signer_updated {
-                check_validator_tracked_shards(&self.client)
+                let validator_signer =
+                    self.client.validator_signer.get().expect("Validator signer just updated");
```

Comment:
as a sanity check: can `update_result.validator_signer_updated` be true in the case of removing the validator key/validator_key.json file? (in which case this expect call will trigger)?

Reply:
No